api/src/main/java/me/lokka30/treasury/api/common/event/EventCaller.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.api.common.event;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.jetbrains.annotations.NotNull;
+
+class EventCaller<T> {
+
+    private Set<EventSubscriber<T>> subscriptions = new TreeSet<>();
+    private final ExecutorService eventCallThreads;
+
+    EventCaller(Class<T> eventClass) {
+        this.eventCallThreads = Executors.newCachedThreadPool(new ThreadFactory() {
```

Comment:
Are we creating a new thread for every event created? Is it possible to run an event in the main thread or we only allow async events?

Reply:
yes we are creating a new cached thread pool* for each event registered. this is because of the way the event blocking works. no it is not possible to call an event in a "main thread" or handle a subscriber in the "main thread", again because of the way event blocking works.